nautobot/extras/models/models.py

```diff
@@ -644,6 +644,48 @@ def size(self):
             return None
 
 
+#
+# Notes
+#
+
+
+@extras_features("graphql", "webhooks")
+class Notes(BaseModel, ChangeLoggedModel):
+    """
+    Notes allows anyone with proper permissions to add a note to an object.
+    """
+
+    assigned_object_type = models.ForeignKey(to=ContentType, on_delete=models.CASCADE)
+    assigned_object_id = models.UUIDField(db_index=True)
+    assigned_object = GenericForeignKey(ct_field="assigned_object_type", fk_field="assigned_object_id")
+    user = models.ForeignKey(
+        to=settings.AUTH_USER_MODEL,
+        on_delete=models.SET_NULL,
+        related_name="note",
+        blank=True,
+        null=True,
+    )
+    user_name = models.CharField(max_length=150, editable=False)
+    name = models.CharField(max_length=150)
+    slug = AutoSlugField(populate_from="name", max_length=150)
```

Comment:
I don't know that I'm a fan of every Note needing to have a globally unique slug. Natural keys are good to have but it might be better to just have something like a natural key / unique_together constraint on `(assigned_object_type, assigned_object_id, created)` or similar?

Reply:
So, Autoslug ensures that every slug is globally unique. In addition, since the delete/add/edit use slug alone, they would need a unique slug.